components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +70,330 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+                         constant:-kBottomPadding]
+          .active = YES;
+      [self.textLabel.centerYAnchor constraintEqualToAnchor:self.buttonContainerView.centerYAnchor]
+          .active = YES;
+    } else {
+      if (self.hasImage) {
+        [self.imageView.leadingAnchor constraintEqualToAnchor:self.layoutMarginsGuide.leadingAnchor
+                                                     constant:kLeadingPadding]
+            .active = YES;
+        [self.imageView.topAnchor constraintEqualToAnchor:self.layoutMarginsGuide.topAnchor
+                                                 constant:kTopPaddingLarge]
+            .active = YES;
+        [self.textLabel.leadingAnchor constraintEqualToAnchor:self.imageView.trailingAnchor
+                                                     constant:kSpaceBetweenIconImageAndTextLabel]
+            .active = YES;
+        [self.buttonContainerView.topAnchor
+            constraintGreaterThanOrEqualToAnchor:self.imageView.bottomAnchor
+                                        constant:kVerticalSpaceBetweenButtonAndTextLabel]
+            .active = YES;
+      } else {
+        [self.textLabel.leadingAnchor constraintEqualToAnchor:self.layoutMarginsGuide.leadingAnchor
+                                                     constant:kLeadingPadding]
+            .active = YES;
+        [self.buttonContainerView.topAnchor
+            constraintGreaterThanOrEqualToAnchor:self.textLabel.bottomAnchor
+                                        constant:kVerticalSpaceBetweenButtonAndTextLabel]
+            .active = YES;
+      }
+      [self.textLabel.topAnchor constraintEqualToAnchor:self.layoutMarginsGuide.topAnchor
+                                               constant:kTopPaddingLarge]
+          .active = YES;
+      [self.textLabel.trailingAnchor constraintEqualToAnchor:self.layoutMarginsGuide.trailingAnchor
+                                                    constant:-kTrailingPadding]
+          .active = YES;
+      NSLayoutConstraint *buttonContainerTopConstraint = [self.buttonContainerView.topAnchor
+          constraintEqualToAnchor:self.textLabel.bottomAnchor
+                         constant:kVerticalSpaceBetweenButtonAndTextLabel];
+      buttonContainerTopConstraint.priority = UILayoutPriorityDefaultLow;
+      buttonContainerTopConstraint.active = YES;
+      [self.buttonContainerView.leadingAnchor
+          constraintEqualToAnchor:self.layoutMarginsGuide.leadingAnchor
+                         constant:kLeadingPadding]
+          .active = YES;
+      [self.buttonContainerView.trailingAnchor
+          constraintEqualToAnchor:self.layoutMarginsGuide.trailingAnchor
+                         constant:-kTrailingPadding]
+          .active = YES;
+      [self.buttonContainerView.bottomAnchor
+          constraintEqualToAnchor:self.layoutMarginsGuide.bottomAnchor
+                         constant:-kBottomPadding]
+          .active = YES;
```

Comment:
I don't see it in this code, but I'd expect the existing constraints to be removed at some point or at least disabled. Did you find that wasn't necessary when the layout style changed?

Reply:
Good catch! That part was missing from this PR. I will work on adding it when refactoring this code block following up the comment above.

Reply:
Addressed in https://github.com/material-components/material-components-ios/commit/63130a82404872e1085ad6201c4100a5de31e84f#diff-9f546e78faa17de3d67395c6ef3629b2R306.